httpbp/middlewares.go

```diff
@@ -510,3 +513,49 @@ func (rr *responseRecorder) WriteHeader(code int) {
 	rr.ResponseWriter.WriteHeader(code)
 	rr.responseCode = code
 }
+
+type wrappedHijacker struct {
+	http.ResponseWriter
+	http.Hijacker
+}
+
+type wrappedFlusher struct {
+	http.ResponseWriter
+	http.Flusher
+}
+
+type wrappedFlushHijacker struct {
+	http.ResponseWriter
+	http.Flusher
+	http.Hijacker
+}
+
+func allowFlushHijack(original, rw http.ResponseWriter) http.ResponseWriter {
+	flusher, isFlusher := original.(http.Flusher)
+	hijacker, isHijacker := original.(http.Hijacker)
+	switch {
+	case isFlusher && isHijacker:
+		return &wrappedFlushHijacker{rw, flusher, hijacker}
+	case isFlusher:
+		return allowFlush(original, rw)
+	case isHijacker:
+		return allowHijack(original, rw)
+	default:
+		return rw
+	}
+
+}
```

Comment:
🔕 it's manage-able as we currently have 2 optional interfaces, it will become a nightmare if there's a 3rd one added in the future. but I also can't find a better way to do this.

Reply:
Yeah this approach is definitely verbose but I can't think of a better one. The only other interface on `http` that a `http.ResponseWriter` could implement would be [`http.Pusher`](https://pkg.go.dev/net/http#Pusher). I don't think it's worth supporting it now as I don't think we have any use case driving it, but that would be a 3rd interface that may come up in the future.

Reply:
This is a quietly known issue amongst the ecosystem, and the approach in the generic case seems to be code generation.

https://www.doxsey.net/blog/fixing-interface-erasure-in-go/